### PR TITLE
docs: add blog post for js-ipfs@0.50.0

### PR DESCRIPTION
**Pull request by achingbrain**
I've set the date to publish on Monday.

### CODE REVIEW

File: content/post/108-js-ipfs-0.50.0.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+This does not come without a price though, maintaining multiple connections can be resource-intensive and in some cases the browser will limit the number of concurrent connections you can have have.
+
+This can be a problem in web browsers if the user opens your app in two tabs, suddenly you have two nodes running with twice the number of open connections.  Worse, they are sharing a datastore and the same peer ID.
+
+Help is at hand in the shape of the [ipfs-message-port-client](https://www.npmjs.com/package/ipfs-message-port-client) and [ipfs-message-port-server](https://www.npmjs.com/package/ipfs-message-port-server) which allow you to run one IPFS node in a [SharedWorker](https://developer.mozilla.org/en-US/docs/Web/API/SharedWorker) and share it between multiple tabs within your application.
+
+There will be a more in depth post here on this subject soon but in the meantime check out the [browser-sharing-node-across-tabs](https://github.com/ipfs/js-ipfs/tree/master/examples/browser-sharing-node-across-tabs) for an example of how to use it!
```
Comment (autonome) [inline content/post/108-js-ipfs-0.50.0.md:29]: s/check out the/check out/

File: content/post/108-js-ipfs-0.50.0.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+`js-ipfs@0.50.0` has changed the default storage of pins to use the datastore instead of a [DAG] and has seen a corresponding speedup when adding very large numbers of files:
+
+<p style="max-width:1000px;margin-left:auto;margin-right:auto;">
+  <img src="/108-js-ipfs-0.50.0/pinning-performance">
+</p>
+
+In the diagram above you can see that as the number of pinned items increases, so does the time it takes to add the next pin.  There's a steep increase at 8,192 pins, which is when the first bucket is considered full and multiple buckets are created which then involves more operations to add the next pin.
+
+js-ipfs@0.47.0 exhibits this very severly, go-ipfs@0.6.0 less so but the performance degredation is still visible.
```
Comment (autonome) [inline content/post/108-js-ipfs-0.50.0.md:51]: typo: severely

File: content/post/108-js-ipfs-0.50.0.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+This does not come without a price though, maintaining multiple connections can be resource-intensive and in some cases the browser will limit the number of concurrent connections you can have have.
+
+This can be a problem in web browsers if the user opens your app in two tabs, suddenly you have two nodes running with twice the number of open connections.  Worse, they are sharing a datastore and the same peer ID.
+
+Help is at hand in the shape of the [ipfs-message-port-client](https://www.npmjs.com/package/ipfs-message-port-client) and [ipfs-message-port-server](https://www.npmjs.com/package/ipfs-message-port-server) which allow you to run one IPFS node in a [SharedWorker](https://developer.mozilla.org/en-US/docs/Web/API/SharedWorker) and share it between multiple tabs within your application.
+
+There will be a more in depth post here on this subject soon but in the meantime check out the [browser-sharing-node-across-tabs](https://github.com/ipfs/js-ipfs/tree/master/examples/browser-sharing-node-across-tabs) for an example of how to use it!
+
+## 📌 Pins in the datastore
```
Comment (autonome) [inline content/post/108-js-ipfs-0.50.0.md:31]: The title of the post mentions ipfs.add() but it's not explicitly tied into this paragraph. Might be worth adding a sentence or somehow just making super explicit that add() does an implicit pin (if that's the case).

File: content/post/108-js-ipfs-0.50.0.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+js-ipfs@0.47.0 exhibits this very severly, go-ipfs@0.6.0 less so but the performance degredation is still visible.
+
+The performance of the approach taken by js-ipfs@0.50.0 compares very favourably and is essentially only limited by the peformance of the underlying datastore.
+
+## 🍫 Uint8Arrays
+
+In the beginning there were Arrays.  Simple arrays that could hold all sorts of mixed types, could not be optimised very well and were an abstraction over blocks of memory.
+
+Then Node.js came along introduced the [Buffer](https://nodejs.org/api/buffer.html) - suddenly JavaScript developers could access memory (sort of) directly! These things held numbers with an integer value range of 0-255 and were blazingly fast.  JavaScript was starting to look like a proper language that you could do resource intensive work in.
```
Comment (autonome) [inline content/post/108-js-ipfs-0.50.0.md:59]: s/along/along and/

File: content/post/108-js-ipfs-0.50.0.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+* Store pins in datastore instead of a DAG ([#2771](https://github.com/ipfs/js-ipfs/issues/2771)) ([64b7fe4](https://github.com/ipfs/js-ipfs/commit/64b7fe41738cbe96d5a9075f0c01156c6f889c40))
+* Add protocol list to ipfs id output ([#3250](https://github.com/ipfs/js-ipfs/issues/3250)) ([1b6cf60](https://github.com/ipfs/js-ipfs/commit/1b6cf600a6b1348199457ca1fe6f314b6eff8c46))
+* IPNS publish in browser example ([#3207](https://github.com/ipfs/js-ipfs/issues/3207)) ([91faec6](https://github.com/ipfs/js-ipfs/commit/91faec6e3d89b0d9883b8d7815c276d44048e739))
+* Update hapi to v20 ([#3245](https://github.com/ipfs/js-ipfs/issues/3245)) ([1aeef89](https://github.com/ipfs/js-ipfs/commit/1aeef89c73f42a2f6cceb7f0598400141ce40e23))
+* Update to libp2p@0.29.0 ([63d4d35](https://github.com/ipfs/js-ipfs/commit/63d4d353c606e4fd487811d8a0014bb2173f11be))
+
+# 🏗 API Changes
+
+## Core API & HTTP API Client
```
Comment (autonome) [inline content/post/108-js-ipfs-0.50.0.md:104]: If these are backwards incompatible changes and require developers to update their code, should call that very clearly. A big shouty banner or whatnot. Something un-ignorable.

**Review by autonome — APPROVED** [review]
looks great! a few changes minor things recommended in the comments

File: content/post/108-js-ipfs-0.50.0.md (hunk 5 above)
Comment (achingbrain) [reply content/post/108-js-ipfs-0.50.0.md:106]: Done!

File: content/post/108-js-ipfs-0.50.0.md (hunk 1 above)
Comment (achingbrain) [reply content/post/108-js-ipfs-0.50.0.md]: Done!

File: content/post/108-js-ipfs-0.50.0.md (hunk 2 above)
Comment (achingbrain) [reply content/post/108-js-ipfs-0.50.0.md]: Done!

File: content/post/108-js-ipfs-0.50.0.md (hunk 3 above)
Comment (achingbrain) [reply content/post/108-js-ipfs-0.50.0.md]: I've changed the title to be more about pinning.

File: content/post/108-js-ipfs-0.50.0.md (hunk 4 above)
Comment (achingbrain) [reply content/post/108-js-ipfs-0.50.0.md]: Done!

**Pull request merged**